rotkehlchen/globaldb/asset_updates/manager.py

```diff
@@ -837,3 +674,51 @@ def _retrieve_update_files(
             }
 
         return updates
+
+    def apply_pending_compatible_updates(self) -> None:
+        """Apply any pending asset updates that are compatible with the current DB version
+        before upgrading to target_db_version.
+
+        This ensures we don't miss any asset updates that would become incompatible
+        after the DB upgrade.
+        """
+        try:
+            max_compatible_version = None
+            info_json = self._get_remote_info_json()
+            latest_assets_version = info_json['latest']
+            current_db_version = self.globaldb.get_schema_version()
+            if current_db_version < FIRST_GLOBAL_DB_VERSION_WITH_COLLECTIONS:
+                return
+
+            start_version = max(FIRST_VERSION_WITH_COLLECTIONS, self.local_assets_version + 1)
+            # Find the highest compatible assets version for current DB version
+            for version in range(start_version, latest_assets_version + 1):
+                update_info = info_json['updates'][str(version)]
+                if update_info['min_schema_version'] <= current_db_version <= update_info['max_schema_version']:  # noqa: E501
+                    max_compatible_version = version
+                elif update_info['min_schema_version'] > current_db_version:
+                    break  # Stop at first incompatible version
+
+            # Apply updates up to the highest compatible version
+            if max_compatible_version is not None:
+                log.debug(
+                    'Found compatible assets version. Updating assets',
+                    from_version=self.local_assets_version,
+                    to_version=max_compatible_version,
+                    global_db_version=current_db_version,
+                )
+
+                # get all the assets in the db and choose remote.
+                with self.globaldb.conn.read_ctx() as cursor:
+                    conflicts: dict[Asset, Literal['remote', 'local']] = {
+                        Asset(row[0]): 'remote'
+                        for row in cursor.execute('SELECT identifier FROM assets')
+                    }
```

Comment:
wait this is quite heave. Can't you do `defaultdict(lambda: 'remote')` here/

Reply:
thanks. i had to update its usage since `.get()` does not use `__getitem__`